src/planner.c

```diff
@@ -300,7 +325,9 @@ timescaledb_set_rel_pathlist(PlannerInfo *root, RelOptInfo *rel, Index rti, Rang
 			switch (nodeTag(*pathptr))
 			{
 				case T_AppendPath:
-					if (should_optimize_append(*pathptr))
+					if (should_chunk_append(*pathptr))
+						*pathptr = ts_chunk_append_path_create(root, rel, ht, *pathptr, false, NIL);
+					else if (should_optimize_append(*pathptr))
```

Comment:
so if the `if (!ordered || !pathkeys_contained_in(root->sort_pathkeys, subpath->pathkeys))` branch is hit in `ts_chunk_append_path_create`, then constraint aware append is still not used. Is that the right behaviour?

Reply:
changed